bee-storage/bee-storage-rocksdb/src/access/truncate.rs

```diff
@@ -0,0 +1,88 @@
+// Copyright 2020-2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    column_families::*,
+    storage::{Storage, StorageBackend},
+};
+
+use bee_ledger::{
+    snapshot::info::SnapshotInfo,
+    types::{Balance, ConsumedOutput, CreatedOutput, LedgerIndex, OutputDiff, Receipt, TreasuryOutput, Unspent},
+};
+use bee_message::{
+    address::{Address, Ed25519Address},
+    milestone::{Milestone, MilestoneIndex},
+    output::OutputId,
+    payload::indexation::PaddedIndex,
+    Message, MessageId,
+};
+use bee_storage::access::Truncate;
+use bee_tangle::{
+    metadata::MessageMetadata, solid_entry_point::SolidEntryPoint, unconfirmed_message::UnconfirmedMessage,
+};
+
+fn truncate(storage: &Storage, cf_str: &'static str) -> Result<(), <Storage as StorageBackend>::Error> {
+    let cf_handle = storage.cf_handle(cf_str)?;
+
+    let mut iter = storage.inner.raw_iterator_cf(cf_handle);
+
+    // Seek to the first key.
+    iter.seek_to_first();
+    // Grab the first key if it exists.
+    let first = if let Some(first) = iter.key() {
+        first.to_vec()
+    } else {
+        // There are no keys to remove.
+        return Ok(());
+    };
+
+    iter.seek_to_last();
+    // Grab the last key if it exists.
+    let last = if let Some(last) = iter.key() {
+        let mut last = last.to_vec();
+        // `delete_range_cf` excludes the last key in the range so a byte is added to be sure the last key is included.
+        last.push(u8::MAX);
+        last
+    } else {
+        // There are no keys to remove.
+        return Ok(());
+    };
+
+    storage.inner.delete_range_cf(cf_handle, first, last)?;
+
+    Ok(())
+}
+
+macro_rules! impl_truncate {
+    ($key:ty, $value:ty, $cf:expr) => {
+        #[async_trait::async_trait]
+        impl Truncate<$key, $value> for Storage {
+            async fn truncate(&self) -> Result<(), <Self as StorageBackend>::Error> {
+                truncate(self, $cf)
```

Comment:
why not insert the body of truncate here directly?. Maybe we should do something like this for most operations as they are almost the same.

Reply:
It's not really possible for all operations. Truncate is straightforward as you don't need a key not a value. Other operations do and the way you handle keys and values varies from an operation to another :/

Reply:
And regarding putting putting the body in the macro, idk. That would generate a lot of code and truncate is not a heavy-required operation so I guess the cost of calling the function is fine. What do you think ?